core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -1184,6 +1184,29 @@ public static int validateUtf8MultiByte(long lo, long hi, byte b) {
         return validateUtf8Decode4Bytes(lo, hi, b);
     }
 
+    /**
+     * Returns up to 6 initial bytes of the given UTF-8 sequence (less if it's shorter)
+     * packed into a zero-padded long value, in little-endian order. This prefix is
+     * stored inline in the auxiliary vector of a VARCHAR column, so asking for it is a
+     * matter of optimized data access. This is not a general access method, it
+     * shouldn't be called unless looking to optimize the access of the VARCHAR column.
+     *
+     * @param seq UTF8 sequence
+     * @return up to 6 initial bytes
+     */
+    public static long zeroPaddedSixPrefix(@NotNull Utf8Sequence seq) {
+        final int size = seq.size();
+        if (size >= Long.BYTES) {
+            return seq.longAt(0) & VARCHAR_INLINED_PREFIX_MASK;
+        }
+        final long limit = Math.min(size, VARCHAR_INLINED_PREFIX_BYTES);
+        long result = 0;
+        for (int i = 0; i < limit; i++) {
+            result |= (seq.byteAt(i) & 0xffL) << (8 * i);
```

Comment:
if perf critical, perhaps the loop can be unrolled

Reply:
We cache the prefix where possible, in SQL functions and now in immutable `Utf8Sequence`s. Unrolling would help if we knew that the sequence is large enough, but unfortunately that's not always the case.